### PR TITLE
chore: update PR template with info about conventional commits

### PR DESCRIPTION
**Pull request by Aflynn50**
Update the PR description template to include note about conventional commits.

## Links

**Jira card:** [JUJU-6153](https://warthogs.atlassian.net/browse/JUJU-6153)

[JUJU-6153]: https://warthogs.atlassian.net/browse/JUJU-6153?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+<!-- 
+The PR title should match: <type>(optional <scope>): <description>.
+
+Please also ensure all commits in this PR comply with our conventional commits specification:
+https://docs.google.com/document/d/1SYUo9G7qZ_jdoVXpUVamS5VCgHmtZ0QA-wZxKoMS-C0
```
Comment (SimonRichardson) [inline PULL_REQUEST_TEMPLATE.md:5]: People outside of Canonical can't view this so it's useless for an open source project. Point this to the main website.

Comment (Aflynn50) [reply PULL_REQUEST_TEMPLATE.md:5]: This doc is actually fully public, the tag on it says its open to everyone, canonical or not

Comment (SimonRichardson) [reply PULL_REQUEST_TEMPLATE.md:5]: Ah ok, go for it.

**Review by SimonRichardson — APPROVED** [review]

File: PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (jnsgruk) [reply PULL_REQUEST_TEMPLATE.md:5]: This could actually be transplanted into the dev docs? https://juju.is/docs/dev

Comment (SimonRichardson) [reply PULL_REQUEST_TEMPLATE.md:5]: @tmihoc :point_up:

**Pull request merged**